app/sidekiq/evss/disability_compensation_form/submit_form8940.rb

```diff
@@ -7,22 +7,56 @@ module DisabilityCompensationForm
     class SubmitForm8940 < Job
       extend Logging::ThirdPartyTransaction::MethodWrapper
 
-      STATSD_KEY_PREFIX = 'worker.evss.submit_form8940'
+      STATSD_KEY = 'worker.evss.submit_form8940.exhausted'
 
-      # Sidekiq has built in exponential back-off functionality for retrys
+      # Sidekiq has built in exponential back-off functionality for retries
       # A max retry attempt of 10 will result in a run time of ~8 hours
       # This job is invoked from 526 background job
-      RETRY = 10
+      sidekiq_options retry: 10
 
-      sidekiq_options retry: RETRY
-
-      # This callback cannot be tested due to the limitations of `Sidekiq::Testing.fake!`
       sidekiq_retries_exhausted do |msg, _ex|
-        Rails.logger.send(
-          :error,
-          "Failed all retries on SubmitForm8940 submit, last error: #{msg['error_message']}"
+        job_id = msg['jid']
+        error_class = msg['error_class']
+        error_message = msg['error_message']
+        timestamp = Time.now.utc
+        form526_submission_id = msg['args'].first
+
+        form_job_status = Form526JobStatus.find_by(job_id:)
+        bgjob_errors = form_job_status.bgjob_errors || {}
+        new_error = {
+          "#{timestamp.to_i}": {
+            caller_method: __method__.to_s,
+            error_class:,
+            error_message:,
+            timestamp:,
+            form526_submission_id:
+          }
+        }
+        form_job_status.update(
+          status: Form526JobStatus::STATUS[:exhausted],
+          bgjob_errors: bgjob_errors.merge(new_error)
+        )
+
+        StatsD.increment(STATSD_KEY)
+
+        ::Rails.logger.warn(
+          'Submit Form 8940 Retries exhausted',
+          { job_id:, error_class:, error_message:, timestamp:, form526_submission_id: }
+        )
+      rescue => e
+        ::Rails.logger.error(
+          'Failure in SubmitForm8940#sidekiq_retries_exhausted',
+          {
+            messaged_content: e.message,
+            job_id:,
+            submission_id: form526_submission_id,
+            pre_exhaustion_failure: {
+              error_class:,
+              error_message:
+            }
+          }
```

Comment:
Love the additional details in the error message. I'm just curious if we could refactor this into a separate module to be used for these 3 forms? Sorry if I'm missing something about the feasibility of that.

Reply:
That was discussed, and ultimately we decided to keep it dumb.  There is actually already a module being used in some of these hooks, but some weird littles prevent it for working in all of them.  I think that would be an excellent itteration tho : )